src/sentry/search/eap/trace_metrics/config.py

```diff
@@ -21,50 +24,123 @@ class TraceMetricsSearchResolverConfig(SearchResolverConfig):
     metric_type: MetricType | None
     metric_unit: str | None
 
-    def extra_conditions(self, search_resolver: SearchResolver) -> TraceItemFilter | None:
-        if not self.metric_name or not self.metric_type:
+    def extra_conditions(
+        self,
+        search_resolver: SearchResolver,
+        selected_columns: list[str] | None,
+        equations: list[str] | None,
+    ) -> TraceItemFilter | None:
+        # use the metric from the config first if it exists
+        if extra_conditions := self._extra_conditions_from_metric(search_resolver):
+            return extra_conditions
+
+        # then try to parse metric from the aggregations
+        if extra_conditions := self._extra_conditions_from_columns(
+            search_resolver, selected_columns, equations
+        ):
+            return extra_conditions
+
+        return None
+
+    def _extra_conditions_from_columns(
+        self,
+        search_resolver: SearchResolver,
+        selected_columns: list[str] | None,
+        equations: list[str] | None,
+    ) -> TraceItemFilter | None:
+        selected_metrics: set[Metric] = set()
+
+        if selected_columns:
+            stripped_columns = [column.strip() for column in selected_columns]
+            for column in stripped_columns:
+                match = fields.is_function(column)
+                if not match:
+                    continue
+
+                resolved_function, _ = search_resolver.resolve_function(column)
+                if not isinstance(resolved_function, ResolvedMetricAggregate):
+                    continue
+
+                if not resolved_function.metric_name or not resolved_function.metric_type:
+                    continue
+
+                metric: Metric = (
+                    resolved_function.metric_name,
+                    resolved_function.metric_type,
+                    resolved_function.metric_unit,
+                )
+                selected_metrics.add(metric)
+
+        if not selected_metrics:
             return None
 
-        metric_name, _ = search_resolver.resolve_column("metric.name")
-        if not isinstance(metric_name.proto_definition, AttributeKey):
-            raise ValueError("Unable to resolve metric.name")
+        if len(selected_metrics) > 1:
+            raise InvalidSearchQuery("Cannot aggregate multiple metrics in 1 query.")
```

Comment:
two questions
1. if you can't select multiple metrics in one query why do we need it in the aggregates?
2. shouldn't we dedupe the selected_metrics list if there's multiple aggregates on the same metric?